src/psyclone/gocean1p0.py

```diff
@@ -1411,93 +1412,29 @@ def gen_arg_setter_code(self, parent):
             sub,
             " Set the arguments for the {0} OpenCL Kernel".format(self.name)))
         index = 0
-        # First the boundary values
-        for boundary in boundary_names:
+        # First the PSy-layer kernel arguments
+        for arg in self.arguments.args:
```

Comment:
Could you fuse these two loops to reduce duplication? e.g.:

    for arg in [karg.name for karg in self.arguments.args] + boundary_names:

Reply:
Ah yes, this is much nicer.